src/main/java/meteordevelopment/meteorclient/systems/modules/world/AutoSign.java

```diff
@@ -6,42 +6,293 @@
 package meteordevelopment.meteorclient.systems.modules.world;
 
 import meteordevelopment.meteorclient.events.game.OpenScreenEvent;
-import meteordevelopment.meteorclient.events.packets.PacketEvent;
+import meteordevelopment.meteorclient.gui.GuiTheme;
+import meteordevelopment.meteorclient.gui.widgets.containers.WHorizontalList;
+import meteordevelopment.meteorclient.gui.widgets.pressable.WButton;
+import meteordevelopment.meteorclient.gui.widgets.WWidget;
+import meteordevelopment.meteorclient.gui.utils.StarscriptTextBoxRenderer;
 import meteordevelopment.meteorclient.mixin.AbstractSignEditScreenAccessor;
+import meteordevelopment.meteorclient.settings.*;
 import meteordevelopment.meteorclient.systems.modules.Categories;
 import meteordevelopment.meteorclient.systems.modules.Module;
+import meteordevelopment.meteorclient.utils.misc.MeteorStarscript;
 import meteordevelopment.orbit.EventHandler;
+import meteordevelopment.starscript.Script;
+import meteordevelopment.starscript.StandardLib;
 import net.minecraft.block.entity.SignBlockEntity;
 import net.minecraft.client.gui.screen.ingame.AbstractSignEditScreen;
+import net.minecraft.network.packet.c2s.play.ClientCommandC2SPacket;
+import net.minecraft.network.packet.c2s.play.PlayerInteractBlockC2SPacket;
+import net.minecraft.network.packet.c2s.play.PlayerActionC2SPacket;
 import net.minecraft.network.packet.c2s.play.UpdateSignC2SPacket;
+import net.minecraft.text.Text;
+import net.minecraft.util.Hand;
+import net.minecraft.util.Util;
+import net.minecraft.util.hit.BlockHitResult;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.TimeZone;
 
 public class AutoSign extends Module {
-    private String[] text;
 
-    public AutoSign() {
-        super(Categories.World, "auto-sign", "Automatically writes signs. The first sign's text will be used.");
-    }
+    private boolean applyingBack = false;
+
+    private final SettingGroup front = settings.createGroup("Front");
+    private final SettingGroup back = settings.createGroup("Back");
+    private final SettingGroup other = settings.createGroup("Other");
+
+    private final List<Script> frontScripts = new ArrayList<>();
+    private final List<Script> backScripts = new ArrayList<>();
+
+    // 4 lines at front of sign
+    public final Setting<Boolean> frontEnabled = front.add(new BoolSetting.Builder()
+        .name("front-enabled")
+        .description("Enable auto writing front of sign.")
+        .defaultValue(true)
+        .build()
+    );
+
+    private final Setting<String> line1front = front.add(new StringSetting.Builder()
+        .name("front-line-1")
+        .description("Text in first line.")
+        .defaultValue("{player}")
+        .onChanged(strings -> recompile(front, frontScripts, 1))
```

Comment:
instead of passing the `SettingGroup` and getting the string from the index, you can just pass in `strings`